compiler/src/iree/compiler/Codegen/Common/MathTransformPass.cpp

```diff
@@ -0,0 +1,155 @@
+//
+// Licensed under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#include "iree/compiler/Codegen/Common/Passes.h"
+#include "mlir/Dialect/Math/Transforms/Approximation.h"
+#include "mlir/Dialect/Math/Transforms/Passes.h"
+#include "mlir/IR/PatternMatch.h"
+#include "mlir/Pass/Pass.h"
+#include "mlir/Transforms/GreedyPatternRewriteDriver.h"
+
+namespace mlir::iree_compiler {
+
+/// Deprecated! This flag had buggy/unintentional semantics.
+/// Its original comment said:
+/// ""use native hardware operations instead of polynomial approximation".
+static llvm::cl::opt<bool> clNativeMathPrecision(
+    "iree-codegen-gpu-native-math-precision",
+    llvm::cl::desc("Deprecated! This flag had buggy/unintentional semantics. "
+                   "Its original description said: \"Skip polynomial lowering "
+                   "for math op natively available on GPU.\""),
+    llvm::cl::init(false));
+
+#define GEN_PASS_DEF_MATHTRANSFORMPASS
+#include "iree/compiler/Codegen/Common/Passes.h.inc"
+
+namespace {
+
+static void populateMathFunctionsRewritePatterns(
+    RewritePatternSet &patterns,
+    const std::function<bool(StringRef)> &predicate) {
+  if (predicate(math::TanOp::getOperationName())) {
+    populateExpandTanPattern(patterns);
+  }
+  if (predicate(math::SinhOp::getOperationName())) {
+    populateExpandSinhPattern(patterns);
+  }
+  if (predicate(math::CoshOp::getOperationName())) {
+    populateExpandCoshPattern(patterns);
+  }
+  if (predicate(math::AsinhOp::getOperationName())) {
+    populateExpandAsinhPattern(patterns);
+  }
+  if (predicate(math::AcoshOp::getOperationName())) {
+    populateExpandAcoshPattern(patterns);
+  }
+  if (predicate(math::AtanhOp::getOperationName())) {
+    populateExpandAtanhPattern(patterns);
+  }
+  if (predicate(math::PowFOp::getOperationName())) {
+    populateExpandPowFPattern(patterns);
+  }
+  if (predicate(math::FPowIOp::getOperationName())) {
+    populateExpandFPowIPattern(patterns);
+  }
+  if (predicate(math::Exp2Op::getOperationName())) {
+    populateExpandExp2FPattern(patterns);
+  }
+  if (predicate(math::RoundEvenOp::getOperationName())) {
+    populateExpandRoundEvenPattern(patterns);
+  }
+}
+
+bool predicateRewrite(StringRef name, IREE::HAL::ExecutableTargetAttr target) {
+  (void)target;                // Currently unused.
+  if (clNativeMathPrecision) { // Legacy.
+    if (name == math::Exp2Op::getOperationName() ||
+        name == math::RoundEvenOp::getOperationName()) {
+      return false;
+    }
+  }
+  // Currently enable all non-approximative rewrites.
+  return true;
+}
+
+bool predicateF32Cast(StringRef name, IREE::HAL::ExecutableTargetAttr target) {
+  (void)target;                // Currently unused.
+  if (clNativeMathPrecision) { // Legacy.
+    return false;
+  }
+  StringRef atan{math::AtanOp::getOperationName()};
```

Comment:
nit: I'd prefer not using braced initializer lists to call a constructor. And we prefer using an equal before the open curly brace if we use a braced initializer list when initializing a variable.

See
https://llvm.org/docs/CodingStandards.html#do-not-use-braced-initializer-lists-to-call-a-constructor for more details.

The style guide does not explain the reason well. here is an article from abseil, which mentions few benefits. E.g., we can avoid the most vexing parse, etc. See https://abseil.io/tips/88 for more details.


```suggestion
  StringRef atan = math::AtanOp::getOperationName();
```